### PR TITLE
sct, keyring: specialize errors

### PR DESCRIPTION
**Pull request by tnytown**
#### Summary

See #547, #543.

In the SCT module, we now distinguish between the following kinds of verification errors:
- Invalid (potentially malicious) input signatures,
- Invalid key ID,
- Malformed input or API misuse.

### CODE REVIEW

**Comment by woodruffw** [comment]
+@tetsuo-cpp for review 🙂

File: sigstore/_internal/sct.py [hunk 1]
```diff
@@ -142,13 +146,34 @@ class InvalidSCTError(Error):
...
+
+            Additional context:
+
+            Message: {str(self)}
+            """
+            + ctx
+        )
+
+
+class InvalidSCTKeyError(Error):
```
Comment (tetsuo-cpp) [inline sigstore/_internal/sct.py:163]: Is there a reason we shouldn't subclass from `InvalidSCTError` here?

**Review by tetsuo-cpp — APPROVED** [review]
LGTM! I left one style related question but if you have a good reason for doing this, feel free to merge as is.

File: sigstore/_internal/sct.py (hunk 1 above)
Comment (tnytown) [reply sigstore/_internal/sct.py:163]: Ah, good catch! I must've forgotten to.

**Review by tetsuo-cpp — APPROVED** [review]

**Pull request merged**